### PR TITLE
docs(slack) overhauled setup instructions and screenshots

### PR DESCRIPTION
**Pull request by jayacryl**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **Documentation**
  - Introduced the DataHub Slack App, enabling data discovery, asset previews, and incident management directly within Slack.
  - Added a detailed guide on configuring Slack for notifications in DataHub, including installation, token generation, and integration steps.

### CODE REVIEW

File: docs-website/sidebars.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ module.exports = {
         "docs/managed-datahub/welcome-acryl",
         {
           type: "doc",
-          id: "docs/managed-datahub/saas-slack-setup",
+          id: "docs/managed-datahub/slack/saas-slack-setup",
```
Comment (jayacryl) [inline docs-website/sidebars.js:214]: to preserve old mapping

File: docs/managed-datahub/slack/saas-slack-app.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import FeatureAvailability from '@site/src/components/FeatureAvailability';
+
+# The Slack App
```
Comment (jjoyce0510) [inline docs/managed-datahub/slack/saas-slack-app.md:3]: Minor: We can just say "Slack App"

File: docs/managed-datahub/slack/saas-slack-app.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+<FeatureAvailability saasOnly />
+
+## Overview
+The DataHub Slack App brings several of DataHub's key capabilities directly into your slack experience. For instance, you can search for an asset directly within slack by simply typing `/acryl search revenue_data`.
```
Comment (jjoyce0510) [inline docs/managed-datahub/slack/saas-slack-app.md:8]: You actually don't need the "search" part :) 

Let's put this inside a top-level section called: "Searching for data assets" 

The "Overview" section can give a brief overview of the QUERY (search) and NOTIFICATION (subscription) use cases supported by Slack

We should also document that you can subscribe to assets via Slack. And we can also add a "Manaing data incidents" bit in here that describes how to resolve and reopen incidents via Slack. I can send you some screenshots there as well

File: docs/managed-datahub/slack/saas-slack-setup.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+For now we support sending notifications to
+- Slack Channel Name (e.g. `#troubleshoot`)
+- Slack Channel ID (e.g. `C029A3M079U`)
+- Specific Users (aka Direct Messages or DMs) via user ID 
+
+By default, the Slack app will be able to send notifications to public channels. If you want to send notifications to private channels or DMs, you will need to invite the Slack app to those channels.
+
+## How to find Team ID and Channel ID in Slack
+***Note** we recommend just using the Slack channel name for simplicity (e.g. `#troubleshoot`)*
```
Comment (yoonhyejin) [inline docs/managed-datahub/slack/saas-slack-setup.md:132]: let's use docusaurus admonition for this part 
image

```suggestion
:::note 

we recommend just using the Slack channel name for simplicity (e.g. `#troubleshoot`)

:::
```

**Review by yoonhyejin — DISMISSED** [review]
Overall LGTM, left a nitpick comment

File: docs/managed-datahub/slack/saas-slack-setup.md (hunk 4 above)
Comment (jayacryl) [reply docs/managed-datahub/slack/saas-slack-setup.md:132]: tyyy

File: docs-website/sidebars.js [hunk 5]
```diff
@@ -247,6 +242,20 @@ module.exports = {
...
+        {
+          Slack: [
+            {
+              type: "doc",
+              id: "docs/managed-datahub/slack/saas-slack-setup",
+              className: "saasOnly",
+            },
+            {
+              type: "doc",
+              id: "docs/managed-datahub/slack/saas-slack-app",
```
Comment (yoonhyejin) [inline docs-website/sidebars.js:254]: Not really sure "Slack" or "Slack App" is the best name for these docs.. maybe Slack Integration /  Slack App Features? 
I don't have a good idea for this so I'll defer to you  😂 

image

File: docs/managed-datahub/slack/saas-slack-setup.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+For now, we support sending notifications to
+- Slack Channel Name (e.g. `#troubleshoot`)
+- Slack Channel ID (e.g. `C029A3M079U`)
+- Specific Users (aka Direct Messages or DMs) via user ID 
+
+By default, the Slack app will be able to send notifications to public channels. If you want to send notifications to private channels or DMs, you will need to invite the Slack app to those channels.
+
+## How to find Team ID and Channel ID in Slack
+:::note
+We recommend just using the Slack channel name for simplicity (e.g. `#troubleshoot`)*
```
[the remark this answers is not in the document]
Comment (jayacryl) [reply docs/managed-datahub/slack/saas-slack-setup.md:133]: Ah good catch thanks

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**